bam_ampliconclip.c

```diff
@@ -526,6 +528,49 @@ static inline void swap_bams(bam1_t **a, bam1_t **b) {
 }
 
 
+// Format OA:Z:(RNAME,POS,strand,CIGAR,MAPQ,NM;
+static inline int tag_original_data(bam1_t *orig, kstring_t *oa_tag) {
+    char strand;
+    uint8_t *nm_tag, *old_oa_tag;
+    uint32_t *cigar;
+    int64_t nm = 0;
+    int i;
+
+    ks_clear(oa_tag);
+
+    if ((old_oa_tag = bam_aux_get(orig, "OA"))) {
+        ksprintf(oa_tag, "%s", bam_aux2Z(old_oa_tag));
+    }
+
+    if (orig->core.flag & BAM_FREVERSE)
+        strand = '-';
+    else
+        strand = '+';
+
+    if ((nm_tag = bam_aux_get(orig, "NM"))) {
+        nm = bam_aux2i(nm_tag);
+    }
+
+    ksprintf(oa_tag, "%s,%"PRIhts_pos",%c,", bam_get_qname(orig), orig->core.pos + 1, strand);
+
+    for (i = 0, cigar = bam_get_cigar(orig); i < orig->core.n_cigar; ++i) {
+        kputw(bam_cigar_oplen(cigar[i]), oa_tag);
+        kputc(bam_cigar_opchr(cigar[i]), oa_tag);
+    }
+
+    if (nm_tag) {
+        ksprintf(oa_tag, ",%d,%"PRId64";", orig->core.qual, nm);
+    } else {
+        ksprintf(oa_tag, "%d,;", orig->core.qual);
+    }
+
+    if (old_oa_tag)
+        bam_aux_del(orig, old_oa_tag);
```

Comment:
I think it's too early to call `bam_aux_del()` here.  The new OA tag will only be added if some clipping happens, so removong the old one at this point makes it disappear in the case where nothing was clipped.  You can see this in action by running ampliconclip --original on the `test/ampliconclip/1_original_tag.expected.sam` file.

Reply:
Yes, good catch.  That was leftover from an earlier iteration.